File: filter_domain.py
import sys, re, json

def filter(filterList):
    pattern = "\|\|.+?\^"
    pre = "*://*."
    lines = filterList.read().splitlines()
    domains = []
    for line in lines:
        x = re.search(pattern, line)
        if x:
            domain = x.group(0)
            domain = domain[2:-1]
            domain = pre + domain + "/*"
            domains.append(domain)
    with open("domains.js", "w") as js_file:
        js_file.write("var blocked_domains = %s;" % json.dumps(domains))
    
"""
usage : python3 filter_domain.py filterlist.txt
create domains.js file

"""


def main(argv):
    file = open(argv[1], "r")
    filter(file)


    
if __name__ == "__main__":
    main(sys.argv)

File: filter_domain.js


//called in create_master_list.js
//currently takes an array of strings
//returns another array of strings
function filter(filterlist){
    var whatever = filterlist.split("\n");
    var reg3 = /^\|\|[a-zA-Z0-9./]+?\^\$third-party*$/ //third party pattern
    var reg1 = /^\|\|[a-zA-Z0-9./]+?\.(net|com)\^*$/ //simple domain
    var pre = "*://*."
    var domains = [];

    whatever.forEach(line => {
        y = line.search(reg3); //third party pattern
        if (y != -1){
            domain = line.slice(y+2);
            temp = domain.search(/[a-zA-Z0-9./]+?\^/);
            if(temp != -1){
                domain = domain.slice(temp, domain.indexOf("^"));
                domain = pre + domain + "/*";
                domains.push(domain);
            }
        }else{ //simple domain 
            x = line.search(reg1);
            if (x != -1){
                domain = line.slice(x+2);
                if(domain[domain.length - 1] == '^'){
                    domain = domain.slice(0,-1);
                }
                domain = pre + domain + "/*";
                domains.push(domain);
            }
        }
    });
    return domains;       
}